common/remote/rmtspawn.cpp

```diff
@@ -447,6 +447,21 @@ void CRemoteSlave::run(int argc, char * argv[])
                         byte action;
                         msg.read(action);
 
+                        // <= 7.6.xx clients also send:
+                        //      num passwords followed by those pwds
+                        // >= 7.8.0 clients until now do not send password data
+
+                        // so next unsigned could be either num passwords or netaddr
+                        // password info has been set to 0 for many versions
+                        // if next unsigned is not 0 then assumme it is an IP address
+                        // as IP address will surely have higher bits set and be > 0
+
+                        unsigned numPasswds = 0;
+                        size32_t origPos = msg.getPos();
+                        msg.read(numPasswds);
+                        if (numPasswds != 0)
```

Comment:
Should I add an additional check here on the value of numPasswds and possibly assert() ?